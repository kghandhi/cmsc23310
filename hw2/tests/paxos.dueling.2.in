# Variation on the dueling proposers, where having P1 fail just one
# tick later allows it to have its value accepted instead of P2's.
 
2 3 130
0 PROPOSE 1 42
7 FAIL PROPOSER 1
8 PROPOSE 2 37
15 FAIL PROPOSER 2
15 RECOVER PROPOSER 1
27 FAIL PROPOSER 1
27 RECOVER PROPOSER 2
39 FAIL PROPOSER 2
39 RECOVER PROPOSER 1
51 FAIL PROPOSER 1
51 RECOVER PROPOSER 2
63 FAIL PROPOSER 2
63 RECOVER PROPOSER 1
76 FAIL PROPOSER 1
76 RECOVER PROPOSER 2
100 RECOVER PROPOSER 1
0 END
