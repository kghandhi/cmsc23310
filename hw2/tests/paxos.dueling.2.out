  0:    -> P1  PROPOSE v=42
  1: P1 -> A1  PREPARE n=1
  2: P1 -> A2  PREPARE n=1
  3: P1 -> A3  PREPARE n=1
  4: A1 -> P1  PROMISE n=1 (Prior: None)
  5: A2 -> P1  PROMISE n=1 (Prior: None)
  6: A3 -> P1  PROMISE n=1 (Prior: None)
  7: ** P1 FAILS **
  8:    -> P2  PROPOSE v=37
  9: P2 -> A1  PREPARE n=2
 10: P2 -> A2  PREPARE n=2
 11: P2 -> A3  PREPARE n=2
 12: A1 -> P2  PROMISE n=2 (Prior: None)
 13: A2 -> P2  PROMISE n=2 (Prior: None)
 14: A3 -> P2  PROMISE n=2 (Prior: None)
 15: ** P2 FAILS **
 15: ** P1 RECOVERS **
 15: P1 -> A1  ACCEPT n=1 v=42
 16: P1 -> A2  ACCEPT n=1 v=42
 17: P1 -> A3  ACCEPT n=1 v=42
 18: A1 -> P1  REJECTED n=1
 19: A2 -> P1  REJECTED n=1
 20: A3 -> P1  REJECTED n=1
 21: P1 -> A1  PREPARE n=3
 22: P1 -> A2  PREPARE n=3
 23: P1 -> A3  PREPARE n=3
 24: A1 -> P1  PROMISE n=3 (Prior: None)
 25: A2 -> P1  PROMISE n=3 (Prior: None)
 26: A3 -> P1  PROMISE n=3 (Prior: None)
 27: ** P1 FAILS **
 27: ** P2 RECOVERS **
 27: P2 -> A1  ACCEPT n=2 v=37
 28: P2 -> A2  ACCEPT n=2 v=37
 29: P2 -> A3  ACCEPT n=2 v=37
 30: A1 -> P2  REJECTED n=2
 31: A2 -> P2  REJECTED n=2
 32: A3 -> P2  REJECTED n=2
 33: P2 -> A1  PREPARE n=4
 34: P2 -> A2  PREPARE n=4
 35: P2 -> A3  PREPARE n=4
 36: A1 -> P2  PROMISE n=4 (Prior: None)
 37: A2 -> P2  PROMISE n=4 (Prior: None)
 38: A3 -> P2  PROMISE n=4 (Prior: None)
 39: ** P2 FAILS **
 39: ** P1 RECOVERS **
 39: P1 -> A1  ACCEPT n=3 v=42
 40: P1 -> A2  ACCEPT n=3 v=42
 41: P1 -> A3  ACCEPT n=3 v=42
 42: A1 -> P1  REJECTED n=3
 43: A2 -> P1  REJECTED n=3
 44: A3 -> P1  REJECTED n=3
 45: P1 -> A1  PREPARE n=5
 46: P1 -> A2  PREPARE n=5
 47: P1 -> A3  PREPARE n=5
 48: A1 -> P1  PROMISE n=5 (Prior: None)
 49: A2 -> P1  PROMISE n=5 (Prior: None)
 50: A3 -> P1  PROMISE n=5 (Prior: None)
 51: ** P1 FAILS **
 51: ** P2 RECOVERS **
 51: P2 -> A1  ACCEPT n=4 v=37
 52: P2 -> A2  ACCEPT n=4 v=37
 53: P2 -> A3  ACCEPT n=4 v=37
 54: A1 -> P2  REJECTED n=4
 55: A2 -> P2  REJECTED n=4
 56: A3 -> P2  REJECTED n=4
 57: P2 -> A1  PREPARE n=6
 58: P2 -> A2  PREPARE n=6
 59: P2 -> A3  PREPARE n=6
 60: A1 -> P2  PROMISE n=6 (Prior: None)
 61: A2 -> P2  PROMISE n=6 (Prior: None)
 62: A3 -> P2  PROMISE n=6 (Prior: None)
 63: ** P2 FAILS **
 63: ** P1 RECOVERS **
 63: P1 -> A1  ACCEPT n=5 v=42
 64: P1 -> A2  ACCEPT n=5 v=42
 65: P1 -> A3  ACCEPT n=5 v=42
 66: A1 -> P1  REJECTED n=5
 67: A2 -> P1  REJECTED n=5
 68: A3 -> P1  REJECTED n=5
 69: P1 -> A1  PREPARE n=7
 70: P1 -> A2  PREPARE n=7
 71: P1 -> A3  PREPARE n=7
 72: A1 -> P1  PROMISE n=7 (Prior: None)
 73: A2 -> P1  PROMISE n=7 (Prior: None)
 74: A3 -> P1  PROMISE n=7 (Prior: None)
 75: P1 -> A1  ACCEPT n=7 v=42
 76: ** P1 FAILS **
 76: ** P2 RECOVERS **
 76: P2 -> A1  ACCEPT n=6 v=37
 77: P2 -> A2  ACCEPT n=6 v=37
 78: P2 -> A3  ACCEPT n=6 v=37
 79: A1 -> P2  REJECTED n=6
 80: A2 -> P2  REJECTED n=6
 81: A3 -> P2  REJECTED n=6
 82: P2 -> A1  PREPARE n=8
 83: P2 -> A2  PREPARE n=8
 84: P2 -> A3  PREPARE n=8
 85: A1 -> P2  PROMISE n=8 (Prior: n=7, v=42)
 86: A2 -> P2  PROMISE n=8 (Prior: None)
 87: A3 -> P2  PROMISE n=8 (Prior: None)
 88: P2 -> A1  ACCEPT n=8 v=42
 89: P2 -> A2  ACCEPT n=8 v=42
 90: P2 -> A3  ACCEPT n=8 v=42
 91: A1 -> P2  ACCEPTED n=8 v=42
 92: A2 -> P2  ACCEPTED n=8 v=42
 93: A3 -> P2  ACCEPTED n=8 v=42
 94:
 95:
 96:
 97:
 98:
 99:
100: ** P1 RECOVERS **
100: P1 -> A2  ACCEPT n=7 v=42
101: P1 -> A3  ACCEPT n=7 v=42
102: A1 -> P1  ACCEPTED n=7 v=42
103: A2 -> P1  REJECTED n=7
104: A3 -> P1  REJECTED n=7
105: P1 -> A1  PREPARE n=9
106: P1 -> A2  PREPARE n=9
107: P1 -> A3  PREPARE n=9
108: A1 -> P1  PROMISE n=9 (Prior: n=8, v=42)
109: A2 -> P1  PROMISE n=9 (Prior: n=8, v=42)
110: A3 -> P1  PROMISE n=9 (Prior: n=8, v=42)
111: P1 -> A1  ACCEPT n=9 v=42
112: P1 -> A2  ACCEPT n=9 v=42
113: P1 -> A3  ACCEPT n=9 v=42
114: A1 -> P1  ACCEPTED n=9 v=42
115: A2 -> P1  ACCEPTED n=9 v=42
116: A3 -> P1  ACCEPTED n=9 v=42

P1 has reached consensus (proposed 42, accepted 42)
P2 has reached consensus (proposed 37, accepted 42)
